cmd/apprepository-controller/controller.go

```diff
@@ -607,6 +608,15 @@ func apprepoSyncJobArgs(apprepo *apprepov1alpha1.AppRepository, config Config) [
 		args = append(args, "--tls-insecure-skip-verify")
 	}
 
+	if len(apprepo.Spec.FilterRules.Rules) > 0 {
+		rulesJSON, err := json.Marshal(apprepo.Spec.FilterRules)
+		if err != nil {
+			log.Errorf("Unable to parse filter rules for %s: %v", apprepo.Name, err)
+		} else {
+			args = append(args, "--filter-rules", string(rulesJSON))
```

Comment:
I was a bit worried about passing an entire json through the command line, but it seems the maximum length for parameters is enough (`MAX_ARG_STRLEN` is 32 times the page size, that is, about 128K), so no need to be worried about :P

Reply:
yes, in this case, the rules were too complex to use simple flags so I took this option. 

Another option would be to get the filter info directly from the `apprepo` resource but for Helm repositories, that's retrieved after the list of charts are resolved.